### ISSUE TITLE
modified use of antsMultivariateTemp...2.sh

### ISSUE THREAD

**Issue by nooralsh** [ANTsX/ANTs#645]
Hello!

I was using antsMTC2.sh on a slurm cluster but kept running into walltime issues because my jobs would get throttled after each iteration submission (I have 900+ images in my template and therefore 900+ jobs per iteration..). So I modified my use and made each iteration its own job array and then ran shapeupdatetotemplate.sh in between each iteration. This seemed to work fine however each template appears to be shrinking because as far as I can tell, the edge of the skull in the new iterations are being registered to the edge of the brain in the template (the images are not skull-stripped). In the end, I have a weirdly smushed small template that is not representative of the population.

Each of the 4 iterations, as copied from the job_?_?.sh submission files, uses default settings as follows:
```
N4BiasFieldCorrection -d 3 -b [200] -c [50x50x40x30,0.00000001] -i ${subj}.nii.gz -o ./template0${subj}Repaired.nii.gz -r 0 -s 2 --verbose 1

antsRegistration -d 3 --float 1 --verbose 1 -u 1 -w [0.01,0.99] -z 1 -r [template0.nii.gz,${subj}.nii.gz,1] -t Rigid[0.1]  
\ -m MI[template0.nii.gz,./template0${subj}Repaired.nii.gz,1,32,Regular,0.25] -c [1000x500x250x0,1e-6,10] -f 6x4x2x1 -s 4x2x1x0 -t Affine[0.1]  
\ -m MI[template0.nii.gz,./template0${subj}Repaired.nii.gz,1,32,Regular,0.25] -c [1000x500x250x0,1e-6,10] -f 6x4x2x1 -s 4x2x1x0 -t SyN[0.1,3,0]  
\ -m CC[template0.nii.gz,./template0${subj}Repaired.nii.gz,1,4] -c [100x100x70x20,1e-9,10] -f 6x4x2x1 -s 3x2x1x0 -o ./temp932${subj}0

antsApplyTransforms -d 3 --float 1 --verbose 1 -i ./template0${subj}Repaired.nii.gz -o ./template0${subj}0WarpedToTemplate.nii.gz 
\ -r template0.nii.gz -t ./temp932${subj}01Warp.nii.gz -t ./temp932${subj}00GenericAffine.mat
```

I looked over several pages of antsRegistration advice, have changed some settings here and there to no avail. I don't see any obvious errors or issues in my settings and have previously used antsMTC2.sh successfully on another cluster, without needing to break it down into 4 separate runs.

1. Is my broken-down modified version of the script missing a step? 
2. Should the registration settings be changed to improve the registration or is the best option to include a mask so that the skull is ignored?
3. Should I just make the template out of say 100 images and then just do a regular ol' antsRegistration to warp each subject to the template? I was kind of hoping making a template with all of them would do this for me... 

Any help is appreciated! Thanks!

**Comment by gdevenyi** [comment]
Related to: https://github.com/ANTsX/ANTs/issues/630

**Comment by cookpa** [comment]
Skull aligned to brain usually results from some combination of brain and (brain + skull) images. Are all of your images and your initial template NOT skull stripped?

I guess it's also possible that the shape update is wrong.

I think you will see very similar templates from using all subjects vs random samples of 100 from the population. But if you want to use a very large population, you might try `antsMultivariateTemplateConstruction.sh`, which is less computationally burdensome since it uses the older `ANTS`.

**Comment by cookpa** [comment]
OK. Hard to say too much more without seeing the code and data, but you might want to run 1 iteration on a subset of subjects and see if the registrations are all good. If so, then it is probably the update causing the problem. There might be some wild card matching causing issues.

I've also found that initialization of a template from mass rigid alignment, or especially averaging of the input unaligned images, can be very noisy and lead to bad convergence. It might be worth doing a few iterations with affine registration only to clean this up before allowing deformable registration. The median option `-a 2` can help at this stage, as can smoothing the template between iterations.

**Comment by nooralsh** [comment]
Thanks @cookpa for the advice -- I ran a couple iterations of just affine transformations using all the subjects, then used the resulting template for a few subsequent nonlinear registrations and my template looks great! I also padded all the original images with about 10 voxels all around because I think one of my main issues was a close crop of the template skull that shaved off the back of the head, causing the registration to match the next iteration of uncropped skulls to the edge of the brain instead.

Anyways, everything looks good now and works smoothly! Thanks again.
